src/middleware/identity/smartManagement.js

```diff
@@ -1,12 +1,40 @@
 'use strict';
 
+const _ = require('lodash');
+
+const config = require('../../config');
 const errors = require('../../errors');
+const inventory = require('../../connectors/inventory');
+const queries = require('../../remediations/remediations.queries');
 
-module.exports = function (req, res, next) {
-    if (!req.entitlements.smart_management) {
-        return next(new errors.Forbidden());
-    }
+module.exports = async function (req, res, next) {
+    try {
+        let rhcConnectedSystems = false;
+        if (config.isMarketplace) {
+            const remediation = await queries.get(req.params.id, req.user.account_number, req.user.username);
 
-    next();
-};
+            if (!remediation) {
+                res.sendStatus(404);
+            }
+
+            const systemsIds = _(remediation.issues).flatMap('systems').map('system_id').uniq().sort().value();
```

Comment:
It's not clear to me whether a marketplace host in the playbook is what actually unlocks the smart management features. My understanding was that a marketplace host (not necessarily involved in the playbook) is what unlocks it. I am going to find out.

Reply:
Btw if we get a chance to bake this logic into the entitlements service we should do so. IMO it's not remediations' responsibility to figure out these oddities.

Reply:
@jharting Just pushed up the "fast lane" changes for customers with the smart_management entitlement.  Also wrt the first question, I believe that a marketplace host only unlocks smart management features when it's part of a playbook.